tests/build/test_generate_config.py

```diff
@@ -2,31 +2,294 @@
+            }
+        },
+    )
+
+    generate_config("K64F", "GCC_ARM", program)
+
+    config_text = program.files.cmake_config_file.read_text()
+
+    assert "ENABLE_NEWLINES" in config_text
+
+
+def test_config_param_from_target_processed_with_default_name_mangling(program):
+    generate_config("K64F", "GCC_ARM", program)
+
+    config_text = program.files.cmake_config_file.read_text()
+
+    assert "MBED_CONF_TARGET_XIP_ENABLE=0" in config_text
+
+
+def test_macros_from_lib_collected(program):
+    macros = ["NS_USE_EXTERNAL_MBED_TLS"]
+    create_mbed_lib_json(program.mbed_os.root / "connectivity" / "mbed_lib.json", "nanostack", macros=macros)
+
+    generate_config("K64F", "GCC_ARM", program)
+
+    config_text = program.files.cmake_config_file.read_text()
+
+    for macro in macros:
+        assert macro in config_text
+
+
+def test_macros_from_target_collected(program):
+    generate_config("K64F", "GCC_ARM", program)
+
+    config_text = program.files.cmake_config_file.read_text()
+
+    for macro in TARGET_DATA["macros"]:
+        assert macro in config_text
+
+
+def test_target_labels_collected_as_defines(program):
+    generate_config("K64F", "GCC_ARM", program)
+
+    config_text = program.files.cmake_config_file.read_text()
+
+    for label in TARGET_DATA["labels"] + TARGET_DATA["extra_labels"]:
+        assert f"TARGET_{label}" in config_text
+
+    for feature in TARGET_DATA["features"]:
+        assert f"FEATURE_{feature}=1" in config_text
+
+    for device in TARGET_DATA["device_has"]:
+        assert f"DEVICE_{device}=1" in config_text
+
+    for form_factor in TARGET_DATA["supported_form_factors"]:
+        assert f"TARGET_FF_{form_factor}" in config_text
+
+
+def test_overrides_lib_config_param_from_app(matching_target_and_filter, program):
+    target, target_filter = matching_target_and_filter
+    create_mbed_lib_json(
+        program.mbed_os.root / "mbed_lib.json", "platform", config={"stdio-baud-rate": {"value": 9600}},
+    )
+
+    create_mbed_app_json(program.root, target_overrides={target_filter: {"platform.stdio-baud-rate": 115200}})
+    generate_config(target, "GCC_ARM", program)
+
+    config_text = program.files.cmake_config_file.read_text()
+
+    assert "MBED_CONF_PLATFORM_STDIO_BAUD_RATE=115200" in config_text
+
+
+def test_overrides_target_config_param_from_app(matching_target_and_filter, program):
+    target, target_filter = matching_target_and_filter
+    create_mbed_app_json(program.root, target_overrides={target_filter: {"target.xip-enable": True}})
+
+    generate_config(target, "GCC_ARM", program)
+
+    config_text = program.files.cmake_config_file.read_text()
+
+    assert "MBED_CONF_TARGET_XIP_ENABLE=1" in config_text
+
+
+@pytest.mark.parametrize(
+    "config_param, config_param_value, expected_output",
+    [
+        ("target.c_lib", "super", 'MBED_C_LIB "super"'),
+        ("target.printf_lib", "maximal-printf", 'MBED_PRINTF_LIB "maximal-printf"'),
```

Comment:
Any other `target.` attributes we should test here?

Reply:
These are currently the only two that work. I'll add more in as xfails, raise an issue and fix them when I rework this code.

Reply:
Thanks for raising https://github.com/ARMmbed/mbed-tools/issues/156